service/host_test.go

```diff
@@ -186,6 +189,84 @@ func TestShutdownWithError_ReturnsError(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestHostShutdown_MetricsAndRuntimeCloser(t *testing.T) {
+	sh := &host{
+		closeChan: make(chan Exit, 1), // Indicates service is running
+		serviceCore: serviceCore{
+			metricsCore: metricsCore{
+				scope:            tally.NoopScope,
+				metricsCloser:    testutils.NoopCloser{},
+				runtimeCollector: metrics.NewRuntimeCollector(tally.NoopScope, time.Millisecond),
+			},
+		},
+	}
+	checkShutdown(t, sh, false)
```

Comment:
The shutdown function itself doesnt seem to care about any of these states. These are testing that when the shutdown function is called, that the metrics, runtime, tracer closers are closed correctly and if there is an error, they return the error. This was not covered by tests before.